From c81501653676053e1913ffd4024810506201888e Mon Sep 17 00:00:00 2001
From: Vadim Markovtsev <v.markovtsev@samsung.com>
Date: Fri, 23 May 2014 09:52:53 +0400
Subject: [PATCH 2/5] Fix LineReceiver.dataReceived on data without a delimiter
 on Python 3

self._buffer.split() may return a single result if no delimiter exists
in the buffer, resulting in ValueError: too many values to unpack on
Python 3
---
 twisted/protocols/basic.py | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/twisted/protocols/basic.py b/twisted/protocols/basic.py
index 7ad7c48..cb182bf 100644
--- a/twisted/protocols/basic.py
+++ b/twisted/protocols/basic.py
@@ -555,8 +555,10 @@ def dataReceived(self, data):
             while self._buffer and not self.paused:
                 if self.line_mode:
                     try:
-                        line, self._buffer = self._buffer.split(
-                            self.delimiter, 1)
+                        splitted = self._buffer.split(self.delimiter, 1)
+                        line = splitted[0]
+                        if len(splitted) > 1:
+                            self._buffer = splitted[1]
                     except ValueError:
                         if len(self._buffer) > self.MAX_LENGTH:
                             line, self._buffer = self._buffer, b''
-- 
1.9.3